### ISSUE TITLE
Analyze WAVE reports for the top 200 .uk WordPress sites

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/oik-a11y#1]
### Background

WebAIM's 2025 report on the accessibility of the top 1,000,000 home pages
( https://webaim.org/projects/million/ )
notes that 
- nearly a quarter of the websites in the top million are WordPress websites
- and they have an average of 50 errors
- which is better than those classified as Elementor (avg 51.1)
- and better than those classified as wpBakery (avg 63.5)
- but worse than those classified as Divi (avg 27.6)

In the autumn my ex-boss from IBM (Chris Winter) got me involved in looking at the problems to try to determine
what can be done to improve these figures, and whether WordPress itself can help.

We looked at 5 of the worst performing websites from the top 1 million and made several observations.
Chris posted these observations at 
https://archivesit.org.uk/blog/what-impact-does-wordpress-have-on-the-web/

In my opinion, we didn't provide enough information to enable website owners to 
realise how easily they could understand their accessibility (a11y) issues and to start dealing with them.

For me at least, doing this analysis prompted me to look at the a11y of 
- my sites, 
- and the plugins and themes I've developed

During the process of improving my sites, plugins and themes I started to log the WAVE report summary,
showing how I was able to improve the reports over time.

I've been able to improve the overall AIM score for quite a few web pages, including those in the WordPress admin area (wp-admin), by making minor improvements to my plugins and themes and/or my WordPress content.  

For some pages I've been able to reduce the total number of errors and alerts to 0, achieving an AIM score of 10.
1 is bad, 5 is the average.  See https://wave.webaim.org/aimscore

### Requirements

What I didn't get from WebAIM's report was the curve showing the number of sites having overall AIM scores from 1 to 10.
This issue is to address that perceived need. 

Initially I wanted to produce a graph comparing the grouping for the top n WordPress sites vs the top n sites that aren't WordPress.
Gathering fresh data manually didn't seem to be a sensible solution.
So I started developing some code that would help me.

The new requirements are:
- for websites with a domain name including `.uk`
- Attempt to determine if it's WordPress or not
- get an AIM report for the home page 
- summarise the reports to produce a chart displaying the summarised data
eg these are just made up numbers.
```
Score,#WordPress,#other
1,1,5
2,5,7
3,10,8
4,15,12
5,20,15
6,25,20
7,20,15
8,10,12
9,8,3
10,1,0
```

<img width="495" height="371" alt="Image" src="https://github.com/user-attachments/assets/23934f84-af3c-4374-bd9f-5a790badf0bf" />

**Comment by bobbingwide** [comment]
> For each domain containing .uk attempt to determine whether or not it's a WordPress website

This turned out to be a little more problematic than I'd expected.
There are several sites which say they can determine if a website is running WordPress.
I tried them with oik-plugins.com. 

Site | Results
---- | -----
https://isititwp.com |  works quite well, reporting that it's WordPress and using genesis-oik theme, which is helpful,  but it doesn't provide an API
https://whatcms.org | only reports that the site is running Nginx - NO good therefore
https://www.themedetect.com  | replied "That's Not A WordPress Site" - NO good therefore
https://cmsdetect.com | reports oik-plugins.com is built with WordPress, but it has annoying adverts

None of them provided a FREE API that I could use.
I ended up modifying the solution I'd used in https://github.com/bobbingwide/oik-monitor/issues/1

7 years ago this solution worked fine for my own sites, but there were problems when I ran it against the TRANCO website list.
For a high percentage of sites I was unable to get a response when attempting to access the home page
with URLs of the following format

https://domain
https://www.domain

where `domain` is the domain name from the TRANCO CSV file. eg `accessable.co.uk`
These domains are reported as:
- `N/A` - when `gethostbyname()` doesn't return an IP or 
- `n/a` when `fetchdomain()` doesn't retrieve any contents. 

Notes: 
- `fetchdomain()` is currently implemented as `file_get_contents()`. 
- For the more recents calls, the last error is included in the files generated in the `wp-versions` folder.
- I tried using `wp_remote_get()` ( implemented as `bw_remote_get()`) but it hung when trying to access `usnews.com`, even though the timeout was set to 15 seconds.

**Comment by bobbingwide** [comment]
> From the output of step 2.
eg
```
oikwp oik-a11y uk 100001 110000
```

In order to find 100 WordPress websites with `.uk` in the domain name
I needed to run step 2 to consider the first 130,000 sites in the TRANCO list.
Eventually
```
oikwp oik-a11y enough
```
- stopped when it had found enough WordPress sites
- where `$enough` is initially set to 100

My routine found
- 100 WordPress websites
- 789 it could access but which weren't WordPress
- 588 sites it couldn't access.

**Comment by bobbingwide** [comment]
### Step 3.

Produce the WebAIM report for each WordPress site listed in `wp-sites.csv`
```
oikwp oik-a11y.php wave wp-sites.csv
```

### Manual process
Before attempting to automate this I produced the WAVE report using eg
https://wave.webaim.org/report#/google.com

Then recorded the summary details in a spreadsheet.

**Comment by bobbingwide** [comment]
### Step 3b. 

Produce the WebAIM reort for eac WordPress site listed in `other_sites.csv`

```
oikwp oik-a11y.php wave wp-sites.csv
```

**Comment by bobbingwide** [comment]
### Steps 3b, 4b


Produce the WebAIM report for each WordPress site listed in `other_sites.csv`

```
oikwp oik-a11y.php wave other_sites.csv
oikwp oik-a11y.php reports other_sites.csv
```


### Steps 2, 3, and 4 for 200 UK WordPress sites vs Other sites

Keep running the code to check which sites are UK WordPress sites vs UK other sites until we've found 200
```
oikwp oik-a11y uk 130001 140000
oikwp oik-a11y enough 
```

**Comment by bobbingwide** [comment]
> Request the WebAIM report in JSON format, using `reporttype=1`. 

Report type 1 isn't detailed enough to drill down into the actuals. 
In order to produce the chart for the top 200 I've chosen to increase the report type parameter to 2.
